bluesky_widgets/qt/search_results.py

```diff
@@ -34,20 +40,22 @@ class DataLoader(QThread):
       viewer from the data cache.
     """
 
-    def __init__(self, queue, get_data, data, data_changed, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self._queue = queue
+    def __init__(self, get_data, data, *args, **kwargs):
+        super().__init__(*args, SignalsClass=DataLoaderSignals, **kwargs)
         self._get_data = get_data
         self._data = data
```

Comment:
Looks like self._data is being shared across two threads (DataLoader) and GuiThread.  Are there guarantees that both threads won't touch data simultaneously? (Maybe I missed something, just confirming to be sure.)

Reply:
1. The GUI thread mutates `_data` by adding a placeholder string (`"..."`).
2. The GUI thread pushes work to the DataLoader thread.
3. Sometime later, the DataLoader thread mutates `_data` again, replacing the placeholder with the actual value.

So far, so good! However, your comment has made me realize that this tidy story breaks down when the catalog is reloaded. At that point `_data` is cleared on the main thread, but there might still be work in the DataLoader's queue. In that scenario, stale values could  get filled in and the true values would never be loaded. This is a critical problem.